### PR TITLE
[POC] Added Predict and SearchRequest workflow step

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/flowframework/workflow/SearchRequestStep.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                String errorMessage = "Failed to search on the index";
+                logger.error(errorMessage, exception);
+                searchIndexFuture.onFailure(new WorkflowStepException(errorMessage, ExceptionsHelper.status(exception)));
+            }));
+
+        } catch (Exception e) {
+            searchIndexFuture.onFailure(e);
+        }
+
+        return null;
```
Comment (joshpalis) [inline src/main/java/org/opensearch/flowframework/workflow/SearchRequestStep.java:115]: We'll need to return the future here

File: src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java [hunk 2]
```diff
@@ -226,6 +231,20 @@ public enum WorkflowSteps {
...
+
+        /** Predict Step */
+        PREDICT(PredictStep.NAME, List.of(MODEL_ID, INPUT_INDEX, INCLUDES), List.of(VECTORS), Collections.emptyList(), null),
+
+        /** Create Search Request Step */
+        SEARCH_REQUEST(
+            SearchRequestStep.NAME,
+            List.of(INDEX_NAME, CONFIGURATIONS),
+            // temporary for POC
+            List.of("hits"),
```
Comment (joshpalis) [inline src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java:244]: We'll need to update this to search request and search response

**Review by joshpalis — APPROVED** [review]
Looks good to me, just one final comment

File: src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java [hunk 3]
```diff
@@ -226,6 +232,20 @@ public enum WorkflowSteps {
...
+
+        /** Predict Step */
+        PREDICT(PredictStep.NAME, List.of(MODEL_ID, INPUT_INDEX, INCLUDES), List.of(VECTORS), Collections.emptyList(), null),
+
+        /** Create Search Request Step */
+        SEARCH_REQUEST_STEP(
+            SearchRequestStep.NAME,
+            List.of(INDEX_NAME, CONFIGURATIONS),
+            // temporary for POC
+            List.of(SEARCH_REQUEST),
```
Comment (joshpalis) [inline src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java:245]: Still needs the search response here

Comment (owaiskazi19) [reply src/main/java/org/opensearch/flowframework/workflow/WorkflowStepFactory.java]: Ah!

**Review by amitgalitz — APPROVED** [review]

**Pull request merged**